x-pack/plugins/lens/public/state_management/lens_slice.ts

```diff
@@ -433,5 +445,61 @@ export const makeLensReducer = (storeDeps: LensStoreDeps) => {
       }
       return newState;
     },
+    [editVisualizationAction.type]: (
+      state,
+      {
+        payload,
+      }: {
+        payload: {
+          visualizationId: string;
+          event: LensEditEvent<keyof LensEditContextMapping>;
+        };
+      }
+    ) => {
+      if (!state.visualization.activeId) {
+        throw new Error('Invariant: visualization state got updated without active visualization');
+      }
+      // This is a safeguard that prevents us from accidentally updating the
+      // wrong visualization. This occurs in some cases due to the uncoordinated
+      // way we manage state across plugins.
+      if (state.visualization.activeId !== payload.visualizationId) {
+        return state;
+      }
+      const activeVisualization = visualizationMap[payload.visualizationId];
+
+      return {
+        ...state,
+        visualization: {
+          ...state.visualization,
+          state: activeVisualization.onEditAction!(state.visualization.state, payload.event),
```

Comment:
PErhaps we could validate the existence of `onEditAction` here as for the visualizationId, and avoid to force cast?

Reply:
Just in case if you're looking at this PR @dej611 - there are still changes coming in. This comment was valid, I didn't think of it :D